### PR TITLE
Testing all the possible GETs the app will have to get

### PR DESCRIPTION
**Pull request by tanadendere**
This PR introduces the api fetches and the interfaces that were created to create types from the API responses.

### CODE REVIEW

File: src/models/chapter.d.ts [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+
+export interface Next {
+  chap: string;
+  vol: unknown;
+  title: unknown;
+  hid: string;
+  lang: string;
+  id: number;
+  href: string;
+}
```
Comment (yanevdw) [inline src/models/chapter.d.ts]: For these ```unknown``` values, if they are optional, you could do ```vol?: string``` for example where the string refers to the type of the parameter if it is not null.

File: src/models/chapter.d.ts [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+export interface Chapter2 {
+  chap: string;
+  vol: unknown;
```
Comment (yanevdw) [inline src/models/chapter.d.ts]: ```Line 116```: Same here

File: src/models/comicChapters.d.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  chapters: Chapter[];
+  total: number;
+  limit: number;
+}
+
+export interface Chapter {
+  id: number;
+  chap: string;
+  title: unknown;
+  vol: unknown;
```
Comment (yanevdw) [inline src/models/comicChapters.d.ts]: Same here.

File: src/models/comicChapters.d.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  lang: string;
+  created_at: string;
+  updated_at: string;
+  up_count: number;
+  down_count: number;
+  is_the_last_chapter: boolean;
+  group_name: string[];
+  hid: string;
+  identities: unknown;
+  md_chapters_groups: MdChaptersGroup[];
```
Comment (yanevdw) [inline src/models/comicChapters.d.ts]: Here as well.

File: src/models/comicInfo.d.ts [hunk 5]
```diff
@@ -0,0 +1,165 @@
+export interface ComicInfo {
+  firstChap: FirstChap;
+  comic: Comic;
+  artists: Artist[];
+  authors: Author[];
+  langList: string[];
+  recommendable: boolean;
+  demographic: unknown;
+  englishLink: unknown;
```
Comment (yanevdw) [inline src/models/comicInfo.d.ts]: Here as well.

File: src/models/comicInfo.d.ts [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+  matureContent: boolean;
+  checkVol2Chap1: boolean;
+}
+
+export interface FirstChap {
+  chap: string;
+  hid: string;
+  lang: string;
+  group_name: string[];
+  vol: unknown;
```
Comment (yanevdw) [inline src/models/comicInfo.d.ts]: Same here.

File: src/models/comicInfo.d.ts [hunk 7]
```diff
@@ -0,0 +1,165 @@
...
+export interface Comic {
+  id: number;
+  hid: string;
+  title: string;
+  country: string;
+  status: number;
+  links: Links;
+  last_chapter: number;
+  chapter_count: number;
+  demographic: unknown;
```
Comment (yanevdw) [inline src/models/comicInfo.d.ts]: Here too.

File: src/models/comicInfo.d.ts [hunk 8]
```diff
@@ -0,0 +1,165 @@
...
+  slug: string;
+  mismatch: unknown;
+  year: number;
+  bayesian_rating: string;
+  rating_count: number;
+  content_rating: string;
+  translation_completed: boolean;
+  chapter_numbers_reset_on_new_volume_manual: boolean;
+  final_chapter: unknown;
+  final_volume: unknown;
```
Comment (yanevdw) [inline src/models/comicInfo.d.ts]: Here too.

**Review by yanevdw — CHANGES_REQUESTED** [review]
Well done! Just check all the unknowns in you models, if they are optional, rather define them as optional.

**Review by yanevdw — APPROVED** [review]
Changes have been implemented. Nicely done!

File: src/models/chapter.d.ts [hunk 9]
```diff
@@ -0,0 +1,143 @@
...
+  chap: string;
+  vol?: number;
+  title?: string;
+  hid: string;
+  lang: string;
+  id: number;
+  href: string;
+}
+
+export interface Chapter2 {
```
Comment (LesediMoko) [inline src/models/chapter.d.ts]: Consider using the same type for Next and Chapter2. They have plenty of properties in common and those that are missing in one or the other can just be optional

File: src/models/searchResult.d.ts [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+
+export interface MuComics {
+  year: number;
+}
+
+export interface MdTitle {
+  title: string;
+}
+
+export interface MdCover {
```
Comment (LesediMoko) [inline src/models/searchResult.d.ts]: this is the second MDCover interface I've come across and they both have the same properties. Maybe just keep one and export when needed

File: src/models/topComic.d.ts [hunk 11]
```diff
@@ -0,0 +1,330 @@
...
+  md_covers: MdCover7[];
+}
+
+export interface MdCover7 {
+  w: number;
+  h: number;
+  b2key: string;
+}
+
+export interface ExtendedNew {
```
Comment (LesediMoko) [inline src/models/topComic.d.ts]: News and ExtendedNews have the same properties. Consider only keeping one

File: src/utilities/api.ts
Comment (LesediMoko) [inline src/utilities/api.ts:1]: consider adding a return type to your functions so that typescript doesn't infer them as 'any'.

**Review by LesediMoko — COMMENTED** [review]
It all looks good. Just made a few comments regarding to the redundancy of your types as well as having a return type for your GET functions

**Review by LesediMoko — CHANGES_REQUESTED** [review]
Great job shaking off the repetitive interface, the code looks really good now! All that's missing are the return types in api.ts for your fetch responses and you'll be good to go 🏄🏽‍♂️

i.e. ` export function getComicInfo(slug: string) : Promise<IComicData> {...} ` 
or just ` export function getComicInfo(slug: string) : IComicData {...} ` if the promise is resolved within the function

**Comment by tanadendere** [comment]
Thanks Lesedi, i had planned to add the add the return types in the next PR because I plan on implementing the API calls a bit differently.

**Review by LesediMoko — APPROVED** [review]
All my concerns have been addressed. Great work, Tana!

**Pull request merged**